core/src/main/java/com/amplifyframework/auth/AuthSignInState.java

```diff
@@ -0,0 +1,114 @@
+//       The team decided to start with these since they are not Cognito specific but consider whether they
+//       should be renamed / supplemented. As future plugins need other states, they would be added here as well.
+public enum AuthSignInState {
+    /**
+     * Next challenge is to supply an SMS_MFA_CODE, delivered via SMS.
+     */
+    SMS_MFA("SMS_MFA"),
+
+    /**
+     *  Next challenge is to supply a SOFTWARE_TOKEN_MFA, delivered via software token.
+     */
+    SOFTWARE_TOKEN_MFA("SOFTWARE_TOKEN_MFA"),
+
+    /**
+     * If MFA is required, users who do not have at least one of the MFA methods set up are presented with this
+     * challenge. The user must set up at least one MFA type to continue to authenticate.
+     */
+    MFA_SETUP("MFA_SETUP"),
+
+    /**
+     * Must select an MFA type. Valid MFA options are SMS_MFA for text SMS MFA, and SOFTWARE_TOKEN_MFA for TOTP
+     * software token MFA.
+     */
+    SELECT_MFA_TYPE("SELECT_MFA_TYPE"),
+
+    /**
+     * Next challenge is to supply PASSWORD_CLAIM_SIGNATURE, PASSWORD_CLAIM_SECRET_BLOCK, and TIMESTAMP
+     * after the client-side SRP calculations.
+     */
+    PASSWORD_VERIFIER("PASSWORD_VERIFIER"),
+
+    /**
+     * This is returned if your custom authentication flow determines that the user should pass another challenge
+     * before tokens are issued.
+     */
+    CUSTOM_CHALLENGE("CUSTOM_CHALLENGE"),
+
+    /**
+     * If device tracking was enabled on your user pool and the previous challenges were passed, this challenge is
+     * returned so that Amazon Cognito can start tracking this device.
+     */
+    DEVICE_SRP_AUTH("DEVICE_SRP_AUTH"),
+
+    /**
+     * Similar to PASSWORD_VERIFIER, but for devices only.
+     */
+    DEVICE_PASSWORD_VERIFIER("DEVICE_PASSWORD_VERIFIER"),
+
+    /**
+     * This is returned if you need to authenticate with USERNAME and PASSWORD directly.
+     * An app client must be enabled to use this flow.
+     */
+    ADMIN_NO_SRP_AUTH("ADMIN_NO_SRP_AUTH"),
+
+    /**
+     * For users which are required to change their passwords after successful first login.
+     * This challenge should be passed with NEW_PASSWORD and any other required attributes.
+     */
+    NEW_PASSWORD_REQUIRED("NEW_PASSWORD_REQUIRED"),
+
+    /**
+     * The flow is completed and no further steps are needed.
+     */
+    DONE("DONE"),
+
+    /**
+     * Unknown sign-in state, potentially unsupported state.
+     */
+    UNKNOWN("UNKNOWN");
+
+    private final String value;
+
+    AuthSignInState(@NonNull String value) {
+        this.value = Objects.requireNonNull(value);
+    }
+
+    @NonNull
+    public String getValue() {
+        return value;
+    }
+
+    @NonNull
+    public static AuthSignInState getEnum(String value) {
+        for (AuthSignInState v : values()) {
+            if (v.getValue().equalsIgnoreCase(value)) {
+                return v;
+            }
+        }
+
+        return AuthSignInState.UNKNOWN;
+    }
+}
```

Comment:
Since the names of your enum identifiers are the same as the contents of their bound `values`, the default `toString()` (which calls `name()`) works, as is.

However, should any enumerated value to added/changed here, this could break in a hard-to-trace way.

I recommend adding a `toString()` impl which emits the `getValue()`, instead of the enum _`name()`_ (the default behavior).

-------------

Actually, your enum values are currently the same as all of your enum identifiers. So you _could_ just skip the inner `value field, and use `name()` for everything. The `fromString(...)` would even still work, you'd just match against `name()`, instead of against `getValue()`. I guess you could keep `getValue()`, and just have that return `name()`, as well. Or, axe `getValue()` and call into the enum via the `name()`.

Reply:
Yeah so I was debating this while writing it and was thinking the flexibility of having a different value than the ENUM name might be worthwhile but I guess that could also just make things more confusing so I'll go ahead and drop it.

Reply:
I could go either way as well!